src/test/java/com/google/sps/tests/NewsTest.java

```diff
@@ -0,0 +1,152 @@
+package com.google.sps.servlets;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mockito;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.sps.data.Article;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.FileReader;
+import java.io.FileInputStream;
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.HashMap;
+
+
+@RunWith(JUnit4.class)
+public final class NewsTest {
+
+  public final static String FULL_HTML_PATH = "src/test/java/com/google/sps/data/fullNewsHTML.txt";
+  public final static String PART_HTML_PATH = "src/test/java/com/google/sps/data/partialNewsHTML.txt";
+  public final static String EMPTY_HTML_PATH = "src/test/java/com/google/sps/data/emptyHTML.txt";
+  public NewsServlet ns;
+  public static List<Article> someArticleList = new ArrayList<>(); 
+  public static List<Article> fullArticleList = new ArrayList<>();
+
+ /**
+  * Helper method that converts a file path of a text file to a String.
+  * @return String version of text in specified file.
+  */
+  private static String textToString(String filePath) throws IOException {
+    StringBuilder stringBuilder = new StringBuilder();
+    try (BufferedReader inputReader = new BufferedReader(new FileReader(filePath))) {
+      String inputLine;
+      while ((inputLine = inputReader.readLine()) != null) {
+        stringBuilder.append(inputLine);
+      }
+    } catch(IOException e) {
+      System.err.println(e);
+      return "";
+    }
+    return stringBuilder.toString();
+  }
+
+ /**
+  * Helper method to check the makeArticleList method in the NewsServlet class.
+  * Checks if makeArticleList on HTML text in the specified file has output equivalent to expectedList.
+  */
+  private void checkMakeArticleList(String filePath, List<Article> expectedList) throws IOException {
+    String htmlString = textToString(filePath);
+    List<Article> articleList = ns.makeArticleList(htmlString);
+    // Check each class instance in each article object is equal.
+    Assert.assertEquals(articleList.size(), expectedList.size());
```

Comment:
[Optional] Theoretically I think there are fancier ways of doing this.  At least in c++ I know there is a way to define an equality method on options and then there is an assertListEquals helper.

You may want to look into junit.Assert which has some particularly fancy helper methods.  See https://stackoverflow.com/questions/3236880/assert-equals-between-2-lists-in-junit for an example.

Reply:
Oh, and I had a typo above, I meant to say "and equality method on objects" not "on options"

Reply:
I tried looking into this for a couple hours, but nothing in the JUnit library or Hamcrest library was working for me.

Reply:
Ok, you can leave it as is but can you just confirm to me that the following doesn't work
`assertThat(articleList, is(expectedList))`

Reply:
Yes, that is not working.

Reply:
I tried Juan's method but it also did not work for Classes instead of Strings.